### DISCUSSION TITLE
How to Use PID Controller with Power_Distribution_Force_Torque_Mode

### DISCUSSION THREAD

**Discussion by JoshProg** [bitcraze/discussions#1189] [Q&A]
Hi all,
I need help on this.

I discovered pid controller was designed to work with Legacy control mode. I want to change it to Force_torque mode in the power distribution script as I am trying new idea with it.
However, the crazyflie didnot rotate, nor the motors move when I did this. 
It is only Bresciani controller that works with the Force_Torque mode, but I prefer the pid controller. 

How can I use this PID controller with other controlMode that is not the Legacy mode that uses roll, pitch and yaw?

I hope to hear from you soon.
Thanks.

**Comment by knmcguire** [comment]
@whoenig, is there someone of your lab that can answer this perhaps? Should this be possible at of this moment?

**Comment by whoenig** [comment]
Each controller is written with a particular output (= control mode). It is possible to map between these modes, but the gains will need to be re-tuned. Apart from using a simulator or have some sort of physical interpretation of the output values (the force/torque mode uses SI units, the legacy mode uses arbitrary values), there is no real advantage in switching the mode.

Our implementation of the Lee controller also uses force/torque, see https://github.com/bitcraze/crazyflie-firmware/pull/1343.

**Reply by JoshProg** [reply]
Thank you.
